vobject/icalendar.py

```diff
@@ -430,12 +430,23 @@ def getrruleset(self, addRDate=False):
                         logging.error('failed to find DUE at all.')
                         return None
 
+                if isinstance(dtstart, datetime.datetime):
+                    dtstart_tzinfo = dtstart.tzinfo
+                else:
+                    dtstart_tzinfo = None
                 if name in DATENAMES:
                     if type(line.value[0]) == datetime.datetime:
-                        list(map(addfunc, line.value))
+                        for dt in line.value:
+                            if dtstart_tzinfo is None:
+                                addfunc(dt.replace(tzinfo=None))
+                            else:
+                                addfunc(dt)
```

Comment:
Please move the test outside the loop. This makes the code more efficient.

Same below.